tests/test_align.py

```diff
@@ -80,6 +80,73 @@ def test_align_47tuc(self):
+                                                 ['ib2u12kaq_flt.fits', 'ib2u12keq_flt.fits',
+                                                  'ib2u12kiq_flt.fits', 'ib2u12klq_flt.fits'],
+                                                 ['ibjt01a1q_flc.fits', 'ibjt01a8q_flc.fits',
+                                                  'ibjt01aiq_flt.fits', 'ibjt01amq_flt.fits',
+                                                  'ibjt01aqq_flt.fits', 'ibjt01auq_flt.fits',
+                                                  'ibjt01yqq_flc.fits', 'ibjt01z0q_flc.fits',
+                                                  'ibjt01zwq_flc.fits', 'ibjt01a4q_flc.fits',
+                                                  'ibjt01acq_flc.fits', 'ibjt01akq_flt.fits',
+                                                  'ibjt01aoq_flt.fits', 'ibjt01asq_flt.fits',
+                                                  'ibjt01avq_flt.fits', 'ibjt01yuq_flc.fits',
+                                                  'ibjt01ztq_flc.fits'],
+                                                 ['ibnh02coq_flc.fits','ibnh02cmq_flc.fits',
+                                                  'ibnh02c7q_flc.fits','ibnh02c5q_flc.fits',
+                                                  'ibnh02cpq_flc.fits','ibnh02c9q_flc.fits',
+                                                  'ibnh02bfq_flc.fits','ibnh02beq_flc.fits']])
+    def test_align_single_visits(self,input_filenames):
+        """ Verify whether single-visit exposures can be aligned to an astrometric standard.
+
+        Characteristics of these tests:
+          * Input exposures include exposures from a number of single visit datasets to explore what impact differing
+            observing modes (differing instruments, detectors, filters, subarray size, etc.) have on astrometry.
+
+        The following datasets are used in these tests:
+
+            * ACS dataset 10048_a1: 2x F344N, 1x F435W, 1x F475W, 2x F502N, 2x F550M, 1x F555W, 1x F606W, 1x F625W,
+              2x F658N, 1x F775W, 1x F814W, 1x F850LP, and 2x F892N ACS/HRC images
+            * ACS dataset 10265_01: 4x F606W full-frame ACS/WFC images
+            * ACS dataset 12580_02: 5x F475W & 6x F814W ACS/WFC images
+            * WFC3 dataset 11663_12: 4x F160W full-frame WFC3/IR images
+            * WFC3 dataset 12219_01: 8x F160W full-frame WFC3/IR images, 9x F336W full-frame WFC3/UVIS images
+            * WFC3 dataset 12379_02: 4X F606W, 4x F502N full-frame WFC3/UVIS images
+
+        """
+        self.input_loc = 'base_tests'
+
+        self.curdir = os.getcwd()
+        for infile in input_filenames:
+            self.get_input_file(infile, docopy=True)
+            updatewcs.updatewcs(infile)
+
+        output_shift_file = 'test_single_visits_shifts.txt'
+        align_to_gaia.align(input_filenames, shift_name=output_shift_file)
+
+        shift_file = Table.read(output_shift_file, format='ascii')
```

Comment:
From experience, you should know there is no guarantee that the table being read (output_shift_file) exists or that the data in the table is useful.  If there is no table, the support level code will throw an exception.  If there is a table, but there were issues aligning to GAIA, then the data is the table is not good.  Again, see the check I do on the table for the x_shift column in align_randomFields to avoid the bad data.

Reply:
Resolved in revision [f327d92d724de8a43435d45bca7f114440227296](https://github.com/spacetelescope/hlapipeline/pull/23/commits/f327d92d724de8a43435d45bca7f114440227296).